lib/ecto/association.ex

```diff
@@ -627,6 +627,12 @@ defmodule Ecto.Association.Has do
       raise ArgumentError, "expected `:where` for #{inspect name} to be a keyword list, got: `#{inspect where}`"
     end
 
+    preload_order = opts[:preload_order] || []
+
+    unless is_list(preload_order) do
```

Comment:
As the has_one and has_many use the same struct, is necessary to validate that the preload_order option is accepted only for `cardinality: :many`?

Reply:
Good question … of course it's up to maintainers, but I guess that if `where` validation looks same then it's all ok.